src/Symfony/Component/OptionsResolver/Tests/OptionsResolverTest.php

```diff
@@ -1019,6 +1019,32 @@ public function testFailIfSetAllowedValuesFromLazyOption()
         $this->resolver->resolve();
     }
 
+    public function testResolveFailsIfInvalidValueFromLazyOption()
```

Comment:
I think you meant `...FromNestedOption` https://symfony.com/doc/current/components/options_resolver.html#nested-options

Lazy options have a different definition https://symfony.com/doc/current/components/options_resolver.html#default-values-that-depend-on-another-option